### PR TITLE
Inital MPLS parsing implementation

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: proto/forwarding/forwarding_common.proto [hunk 1]
```diff
@@ -178,6 +180,9 @@ enum PacketFieldNum {
   PACKET_FIELD_NUM_HOST_PORT_ID = 63; // Host port id (metadata).
   PACKET_FIELD_NUM_L2MC_GROUP_ID = 64; // L2MC Group id (metadata).
   PACKET_FIELD_NUM_POLICER_ID = 65; // Policer id (metadata).
+  PACKET_FIELD_NUM_MPLS_LABEL = 66; // MPLS Label
+  PACKET_FIELD_NUM_MPLS_TC = 67; // MPLS Traffic Class
```
Comment (guoshiuan) [inline proto/forwarding/forwarding_common.proto:184]: Do we also want to add the bottom of stack field?

File: dataplane/forwarding/protocol/mpls/mpls.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+		return true, nil
+	case fwdpb.PacketFieldNum_PACKET_FIELD_NUM_MPLS_TC:
+		field.SetBits(tcBitOffset, tcBitLen, uint64(arg[0]))
+		return true, nil
+	default:
+		return true, field.Set(arg)
+	}
+}
+
+func (m *mpls) Remove(id fwdpb.PacketHeaderId) error {
```
Comment (guoshiuan) [inline dataplane/forwarding/protocol/mpls/mpls.go:136]: id is not referred in the code?

File: dataplane/forwarding/protocol/mpls/mpls.go [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+	if len(m.labels) > 1 {
+		m.labels = m.labels[1:]
+		return nil
+	}
+	m.labels = nil
+	return nil
+}
+
+func (m *mpls) Modify(id fwdpb.PacketHeaderId) error {
```
Comment (guoshiuan) [inline dataplane/forwarding/protocol/mpls/mpls.go:145]: id is not referred in the method. What is modified?

File: dataplane/forwarding/protocol/mpls/mpls.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	m.labels = append([]*mplsLabel{{hdr: make(frame.Header, 4)}}, m.labels...)
+	return nil
+}
+
+func (m *mpls) Rebuild() error {
+	if len(m.labels) == 0 {
+		return nil
+	}
+	// Set the bottom of stack bit 0 on all but the last label. Set that it to 1.
```
Comment (guoshiuan) [inline dataplane/forwarding/protocol/mpls/mpls.go:154]: Can you reword the comment?

File: proto/forwarding/forwarding_common.proto (hunk 1 above)
Comment (DanG100) [reply proto/forwarding/forwarding_common.proto:184]: maybe later, we currently set it automatically

File: dataplane/forwarding/protocol/mpls/mpls.go (hunk 2 above)
Comment (DanG100) [reply dataplane/forwarding/protocol/mpls/mpls.go:136]: id is always mpls for this func

File: dataplane/forwarding/protocol/mpls/mpls.go (hunk 3 above)
Comment (DanG100) [reply dataplane/forwarding/protocol/mpls/mpls.go:145]: modify adds a label to the stack. the id is always mpls

File: dataplane/forwarding/protocol/mpls/mpls.go (hunk 4 above)
Comment (DanG100) [reply dataplane/forwarding/protocol/mpls/mpls.go:154]: done

**Review by guoshiuan — APPROVED** [review]

**Pull request merged**